frame/whitelist/src/lib.rs

```diff
@@ -112,6 +112,24 @@ pub mod pallet {
 		CallAlreadyWhitelisted,
 	}
 
+	#[pallet::hooks]
```

Comment:
I am not familiar enough with these pallets to know why only these two get these integrity check and not others, but my gut feeling is to ask: is there a way to integrate this as a "lint" of some sort in the benchmarking cli? as in: 

The benchmarking cli will conclude a weight as 'fn(x)'. It can then already evaluate this function and assert (possibly as a warning) that f(low) < f(mid) < f(high), for the range of x.

Reply:
Yea, it would be better. As for the makeshift reason: the scheduler pallet is used by parachains to do runtime upgrades. So if its weight is wrong, the chains will be unable to upgrade anymore.  
But i will probably revert it and instead try to do https://github.com/paritytech/substrate/issues/10762.